src/network/local_socket_reply.cpp

```diff
@@ -266,26 +287,57 @@ void mp::LocalSocketReply::read_reply()
 
 void mp::LocalSocketReply::parse_reply()
 {
-    auto reply_lines = reply_data.split('\n');
+    bool chunked_transfer_encoding{false};
+    auto reply_lines = split_http_lines(reply_data);
     auto it = reply_lines.constBegin();
 
     parse_status(*it);
 
+    qint64 data_len{0};
+
     for (++it; it != reply_lines.constEnd(); ++it)
     {
+        bool ok{false};
+
         if ((*it).contains("Transfer-Encoding") && (*it).contains("chunked"))
+        {
             chunked_transfer_encoding = true;
+        }
+        else if ((*it).contains("Content-Length"))
+        {
+            data_len = (*it).split(' ')[1].toLongLong(&ok, 16);
+        }
 
         if ((*it).isEmpty() || (*it).startsWith('\r'))
         {
-            // Advance to the body
-            // Chunked transfer encoding also includes a line with the amount of
-            // bytes (in hex) in the chunk. We just skip it for now.
-            it = chunked_transfer_encoding ? it + 2 : it + 1;
+            ++it;
+
+            if (chunked_transfer_encoding)
+            {
+                QByteArray chunked_data;
+
+                while (true)
+                {
+                    data_len = (*it).toLongLong(&ok, 16);
+
+                    if (data_len == 0)
+                    {
+                        content_data = chunked_data;
+                        return;
+                    }
 
-            content_data = (*it).trimmed();
+                    ++it;
 
-            break;
+                    chunked_data.append((*it).constData(), data_len);
+
+                    ++it;
+                }
+            }
+            else
+            {
+                content_data = (*it).trimmed();
```

Comment:
Not using `data_len` here? You read it from the `Content-Length` header above?

Reply:
Hmm, I suppose instead of just taking the content wholesale, it would be safer (and more correct) to only copy the length specified in the `Content-Length` header.

Reply:
Again, this is a Good Enough™ type of thing in that it's assumed the rest of the body is the content data.  I'll add a comment here explaining that.